frontend/src/providers/Alert.tsx

```diff
@@ -16,15 +14,14 @@ function AlertRuleProvider({
 }: {
 	children: React.ReactNode;
 }): JSX.Element {
-	const [isAlertRuleDisabled, setIsAlertRuleDisabled] = useState<
-		boolean | undefined
-	>(undefined);
-
-	const value = React.useMemo(
-		() => ({ isAlertRuleDisabled, setIsAlertRuleDisabled }),
-		[isAlertRuleDisabled],
+	const [alertRuleState, setAlertRuleState] = useState<string | undefined>(
+		undefined,
```

Comment:
Can you please remove undefined which we are passing explicitly as the initial state?